homeassistant/components/squeezebox/media_player.py

```diff
@@ -460,6 +468,44 @@ async def async_play_media(
             )
             media_id = play_item.url
 
+        if announce:
+            if media_type not in MediaType.MUSIC:
+                raise ServiceValidationError(
+                    "Announcements must have media type of 'music'.  Playlists are not supported"
+                )
+
+            extra = kwargs.get(ATTR_MEDIA_EXTRA, {})
+            cmd = "announce"
+            if raw_announce_volume := extra.get(ATTR_ANNOUNCE_VOLUME):
+                try:
+                    announce_volume = float(raw_announce_volume)
+                    if not (0 < announce_volume <= 1):
+                        raise ServiceValidationError(
+                            f"{ATTR_ANNOUNCE_VOLUME} must be a number greater than 0 and less than or equal to 1"
+                        ) from None
+                except ValueError:
+                    raise ServiceValidationError(
+                        f"{ATTR_ANNOUNCE_VOLUME} must be a number greater than 0 and less than or equal to 1"
+                    ) from None
+                self._player.set_announce_volume(int(announce_volume * 100))
+            else:
+                self._player.set_announce_volume(None)
```

Comment:
Let's refactor this so we're not raising the same error twice and calling `set_announce_volume` twice. The Ruff error in the original suggestion was to `Abstract 'raise' to inner function`. This can look like a module-level function that gets the value and performs the validation:

```python
def get_announce_volume(extra: dict) -> float | None:
    """Get announce volume from extra service data."""
    if ATTR_ANNOUNCE_VOLUME not in extra:
        return None
    announce_volume = float(extra[ATTR_ANNOUNCE_VOLUME])
    if not (0 < announce_volume <= 1):
        raise ValueError
    return announce_volume
```

Then here, this simplifies greatly:
```suggestion
            try:
                announce_volume = get_announce_volume(extra)
            except ValueError as error:
                raise ServiceValidationError(
                     f"{ATTR_ANNOUNCE_VOLUME} must be a number greater than 0 and less than or equal to 1"
                )
            else:
                self._player.set_announce_volume(announce_volume)
```